File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        
        for (String token : tokens) {
            if ("+-/*".contains(token)) {
                int first = stack.pop();
                int second = stack.pop();
                int result = 0;
                switch(token) {
                    case "+":
                        result = second + first;
                        break;
                    case "-":
                        result = second - first;
                        break;
                    case "*":
                        result = second * first;
                        break;
                    case "/":
                        result = second / first;
                        break;
                
                } 
                stack.push(result);
            } else
                stack.push(Integer.valueOf(token));
        }
        
        return stack.peek();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operands_set = {'+', '-', '*', '/'}
        for token in tokens:
            if token in operands_set:
                second_item = stack.pop()
                first_item = stack.pop()
                
                if token == '+':
                    stack.append(first_item + second_item)
                
                elif token == '-':
                    stack.append(first_item - second_item)
                
                elif token == '/':
                    stack.append(int(first_item / second_item))
                
                elif token == '*':
                    stack.append(first_item * second_item)
            else:
                stack.append(int(token))
        
        return stack[-1]
                
                
            
        